storage/scanner.go

```diff
@@ -125,6 +125,23 @@ func (rs *rangeScanner) RemoveRange(rng *Range) {
 	rs.removed <- rng
 }
 
+// paceInterval returns a duration between iterations to allow us to pace
+// the scan.
+func (rs *rangeScanner) paceInterval(start time.Time) time.Duration {
+	elapsed := time.Now().Sub(start)
+	remainingNanos := rs.interval.Nanoseconds() - elapsed.Nanoseconds()
+	if remainingNanos < 0 {
+		remainingNanos = 0
+	}
+	count := rs.iter.EstimatedCount()
+	if count < 1 {
+		count = 1
+	}
+	interval := time.Duration(remainingNanos / int64(count))
+	log.V(6).Infof("pace interval set to %s", interval)
```

Comment:
My other comment got buried, but I think this side effect should be in `scanLoop`, not in this method, which should remain pure.